receiver/libhoneyreceiver/receiver.go

```diff
@@ -261,8 +264,28 @@ func (r *libhoneyReceiver) handleEvent(resp http.ResponseWriter, req *http.Reque
 		return
 	}
 
-	noErrors := []byte(`{"errors":[]}`)
-	writeResponse(resp, enc.ContentType(), http.StatusOK, noErrors)
+	// return clean response if no errors above
+	noErrors := response.MakeResponse([]int{})
+
+	var responseBody []byte
+	var contentType string
+
+	switch enc.ContentType() {
+	case encoder.MsgpackContentType:
+		// For msgpack requests, return msgpack response
+		responseBody, err = msgpack.Marshal(noErrors)
+		contentType = encoder.MsgpackContentType
+	default:
+		// For JSON requests, return JSON response
+		responseBody, err = json.Marshal(noErrors)
+		contentType = encoder.JSONContentType
+	}
+
+	if err != nil {
+		errorutil.HTTPError(resp, err)
+		return
+	}
```

Comment:
Can we check for this error case before creating `noErrors` and doing the switch logic?